docs/content/config/security/ssl.md

```diff
@@ -158,6 +158,92 @@ Certbot provisions certificates to `/etc/letsencrypt`. Add a volume to store the
+
+Obtain a Cloudflare API token:
+
+1. Login into your Cloudflare dashboard.
+2. Navigate to the [API Tokens page](https://dash.cloudflare.com/profile/api-tokens).
+3. Click "Create Token", and choose the `Edit zone DNS` template (_Certbot [requires the `ZONE:DNS:Edit` permission](https://certbot-dns-cloudflare.readthedocs.io/en/stable/#credentials)_).
+
+    !!! warning "Only include the necessary Zone resource configuration"
+    
+        Be sure to configure "Zone Resources" section on this page to `Include -> Specific zone -> <your zone here>`.
+
+        This restricts the API token to only this zone (domain) which is an important security measure.
+
+4. Store the _API token_ you received in a file `cloudflare.ini` with content:
+
+    ```dosini
+    dns_cloudflare_api_token = YOUR_CLOUDFLARE_TOKEN_HERE
+    ```
+
+   - As this is sensitive data, you should restrict access to it with `chmod 600` and `chown 0:0`.
+   - Store the file in a folder if you like, such as `docker-data/certbot/secrets/`.
+5. Your `docker-compose.yml` should include the following:
+
+    ```yaml
+    services:
+      mailserver:
+        environments:
+          # Set SSL certificate type.
+          - SSL_TYPE=letsencrypt
+        volumes:
+          # Mount the cert folder generated by Certbot into mail-server:
+          - ./docker-data/certbot/certs/:/etc/letsencrypt/:ro
+
+      certbot-cloudflare:
+        image: certbot/dns-cloudflare:latest
+        command: certonly --dns-cloudflare --dns-cloudflare-credentials /run/secrets/cloudflare-api-token -d mail.example.com
+        volumes:
+          - ./docker-data/certbot/certs/:/etc/letsencrypt/
+          - ./docker-data/certbot/logs/:/var/log/letsencrypt/
+        secrets:
+          - cloudflare-api-token
+
+    # Docs: https://docs.docker.com/engine/swarm/secrets/#use-secrets-in-compose
+    # WARNING: In compose configs without swarm, the long syntax options have no effect,
+    # Ensure that you properly `chmod 600` and `chown 0:0` the file on disk. Effectively treated as a bind mount.
+    secrets:
+      cloudflare-api-token:
+        file: ./docker-data/certbot/secrets/cloudflare.ini
+    ```
+
+    Alternative using the `docker run` command (`secrets` feature is not available):
+
+      ```sh
+      docker run \
+        --volume "${PWD}/docker-data/certbot/certs/:/etc/letsencrypt/" \
+        --volume "${PWD}/docker-data/certbot/logs/:/var/log/letsencrypt/" \
+        --volume "${PWD}/docker-data/certbot/secrets/:/tmp/secrets/certbot/"
+        certbot/dns-cloudflare \
+        certonly --dns-cloudflare --dns-cloudflare-credentials /tmp/secrets/certbot/cloudflare.ini -d mail.example.com
+      ```
+
+6. Run the service to provision a certificate:
+
+    ```sh
+    docker-compose run certbot-cloudflare
+    ```
```

Comment:
As you've mentioned this only allows to provision the certificate and different steps are required for renewal. Would you like to inform the user about this and how to go about renewal?

Presently pairing the service in a compose config next to the DMS service would run that service and command each time the mail-server is brought up with a `docker-compose up -d`, is that going to be a concern? Should it be treated as a separate compose file?

I've not reviewed the other guides on this page, they may have a similar concern but not one you need to worry about :+1:

---

I [came across a very similar step-by-step guide](https://www.nodinrogers.com/post/2022-03-10-certbot-cloudflare-docker/#3-create-the-docker-composeyml-file) that shows that renewal command is pretty much the same but includes `renew` instead of `certonly`, they also have a few extra options in their commands (_also via a docker-compose file_) that might be worthwhile.

It seems common to just setup a cron or systemd timer on the host to schedule renewals with Certbot?

---

I'm not blocking approval on this, but it would be appreciated if addressed :sweat_smile:

Reply:
Sure, it'd be better if we add that part about renewal, gonna working on this part later and expect to create a commit later today

Reply:
I've completed add documentation for renewing certificate in [8560bdf](https://github.com/docker-mailserver/docker-mailserver/pull/2968/commits/8560bdf7f65c177a1a207104c2841eb2cad6c2be), waiting for github-action to complete building preview